config/routes.rb

```diff
@@ -41,14 +41,18 @@
   # Sets the devise scope to be used in the controller.
   # http://www.rubydoc.info/github/plataformatec/devise/ActionDispatch%2FRouting%2FMapper%3Adevise_scope
   devise_scope :user do
+    get '/users/:id', to: redirect('/users/%{id}/profile')
```

Comment:
One note though: Is this redirect necessary, if all links already link to `/users/:id/profile`?

Reply:
It should not be needed directly. It is convenient, tough (for devs that got used to the url and type it in manually and to keep possible references from the outside intact)